libraries/AP_Airspeed/AP_Airspeed_Health.cpp

```diff
@@ -22,71 +21,76 @@ void AP_Airspeed::check_sensor_ahrs_wind_max_failures(uint8_t i)
         // slow the checking rate
         return;
     }
+    state[i].failures.last_check_ms = now_ms;
 
-    const float aspeed = get_airspeed();
-    const float wind_max = AP::ahrs().get_max_wind();
-
-    if (aspeed <= 0 || wind_max <= 0) {
-        // invalid estimates
+    if (!is_positive(_wind_max)) {
+        // nothing to do
         return;
     }
 
-    state[i].failures.last_check_ms = now_ms;
+    const float aspeed = get_airspeed();
+    if (aspeed <= 0) {
+        // invalid estimate
+        return;
+    }
 
-    // update state[i].failures.health_probability via LowPassFilter
     float speed_accuracy;
     const AP_GPS &gps = AP::gps();
-    if (gps.speed_accuracy(speed_accuracy)) {
-        const float gnd_speed = gps.ground_speed();
-
-        if (aspeed > (gnd_speed + wind_max) || aspeed < (gnd_speed - wind_max)) {
-            // bad, decay fast
-            const float probability_coeff = 0.90f;
-            //state[i].failures.health_probability = probability_coeff*state[i].failures.health_probability + (1.0f-probability_coeff)*0.0f;
-            state[i].failures.health_probability = probability_coeff*state[i].failures.health_probability; // equivalent
-
-        } else if (aspeed < (gnd_speed + wind_max) && aspeed > (gnd_speed - wind_max)) {
-            // good, grow slow
-            const float probability_coeff = 0.98f;
-            state[i].failures.health_probability = probability_coeff*state[i].failures.health_probability + (1.0f-probability_coeff)*1.0f;
-        }
+    if (!gps.speed_accuracy(speed_accuracy)) {
```

Comment:
check for a 3D fix?

Reply:
if no 3D fix then always use airspeed

Reply:
changed to check for at least a 3D fix. If sensor was auto-disabled and GPS goes bad it will now be re-enabled.